system/lib/emmalloc.cpp

```diff
@@ -905,6 +896,16 @@ void emmalloc_free(void *ptr)
   MALLOC_ACQUIRE();
 
   uint32_t size = region->size;
+#ifdef EMMALLOC_DEBUG_LOG
+  if (size < sizeof(Region) || !region_is_in_use(region))
+  {
+    if (debug_region_is_consistent(region))
+      // LLVM wasm backend bug: cannot use MAIN_THREAD_ASYNC_EM_ASM() here, that generates internal compiler error
```

Comment:
I believe this bug can be worked around with having a single EM_ASM and passing into it a string for the initial differing text. As the rest is the same, that would also be shorter here in the code.

Reply:
If I pass it as a string, I need to convert it from Wasm string to JS string, which is kind of dirty... I think I'll leave it like that.